src/runtime_src/core/common/memaccess.h

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (C) 2016-2021 Xilinx, Inc
+ * Copyright (C) 2022 Advanced Micro Devices, Inc. - All rights reserved
+ * 
+ * Author: Sonal Santan
+ * Simple command line utility to inetract with SDX PCIe devices
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+#ifndef MEMACCESS_H
+#define MEMACCESS_H
+
+#include "core/common/device.h"
+
+#include <string>
+
+namespace xrt_core {
+
+// Read from a device's memory banks
+// This function safely reads from a device's memory banks. It will ensure that the read
+// attempts start/end on memory bank borders when applicable. This prevents reading from an unused bank
+// or writing out of bounds. The memory bank usage may not be contiguous and this function account for 
+// these situations.
```

Comment:
Shorten the lines to prevent me from getting cross eyed.
In Emacs I let it break the lines in auto-fill mode (ESC-q) which would produce this:
```suggestion
// Read from a device's memory banks.
//
// This function safely reads from a device's memory banks. It will
// ensure that the read attempts start/end on memory bank borders when
// applicable. This prevents reading from an unused bank or writing
// out of bounds. The memory bank usage may not be contiguous and this
// function account for these situations.
```